pkg/controller/changetierrequest/changetierrequest_controller.go

```diff
@@ -74,5 +85,132 @@ func (r *ReconcileChangeTierRequest) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, nil
+	// if is complete, then check when status was changed delete it if the requested duration has passed
+	completeCond, found := condition.FindConditionByType(changeTierRequest.Status.Conditions, toolchainv1alpha1.ChangeTierRequestComplete)
+	if found && completeCond.Status == corev1.ConditionTrue {
```

Comment:
you could probably combine the 2 statements above into:
```
if completeCond, found := condition.FindConditionByType(changeTierRequest.Status.Conditions, toolchainv1alpha1.ChangeTierRequestComplete);  found && completeCond.Status == corev1.ConditionTrue {
...
}
```

Reply:
I could, but I don't like such long if statements - it's hard to read it. I would rather keep it as it is if you don't mind.

Reply:
ok, fair enough. I prefer the combined style of statements,  but it's a matter of taste ;)